internal/config/entitlements.go

```diff
@@ -0,0 +1,22 @@
+package config
+
+// implements the authorization EntitlementGetter interface
+func (c *Config) GetEntitlements(distrbution string, boundaries ...string) []Entitlement {
```

Comment:
It seems like this could be a simpler interface that an auth provider would need to validate entitlement. 
```
func Entitled(claims []string, req requestSpec) Bool 
```

The middleware parses claims and then just asks whatever meets the Entitler interface for a yes/no answer. That keeps all of the entitlement logic, including knowing what an entitlement looks like out of the http handler.

Reply:
That works.